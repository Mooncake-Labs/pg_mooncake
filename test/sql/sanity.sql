CREATE TABLE t (a int, b text) USING columnstore;
INSERT INTO t VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd'), (5, 'e');
INSERT INTO t VALUES (2, 'f'), (3, 'g'), (4, 'h');
UPDATE t SET b = a + 1 WHERE a > 3;
DELETE FROM t WHERE a < 3;
INSERT INTO t SELECT 2, b FROM t WHERE a = 3;
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a int) USING columnstore;
ANALYZE t;

BEGIN;
INSERT INTO t VALUES (123);
SELECT * FROM t;
ROLLBACK;
SELECT * FROM t;

BEGIN;
INSERT INTO t VALUES (123);
SELECT * FROM t;
COMMIT;
SELECT * FROM t;

DROP TABLE t;
