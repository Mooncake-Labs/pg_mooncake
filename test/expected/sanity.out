CREATE TABLE t (a int, b text) USING columnstore;
INSERT INTO t VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd'), (5, 'e');
INSERT INTO t VALUES (2, 'f'), (3, 'g'), (4, 'h');
UPDATE t SET b = a + 1 WHERE a > 3;
DELETE FROM t WHERE a < 3;
INSERT INTO t SELECT 2, b FROM t WHERE a = 3;
SELECT * FROM t;
 a | b 
---+---
 4 | 5
 5 | 6
 4 | 5
 3 | c
 3 | g
 2 | c
 2 | g
(7 rows)

DROP TABLE t;
CREATE TABLE t (a int) USING columnstore;
ANALYZE t;
BEGIN;
INSERT INTO t VALUES (123);
SELECT * FROM t;
  a  
-----
 123
(1 row)

ROLLBACK;
SELECT * FROM t;
 a 
---
(0 rows)

BEGIN;
INSERT INTO t VALUES (123);
SELECT * FROM t;
  a  
-----
 123
(1 row)

COMMIT;
SELECT * FROM t;
  a  
-----
 123
(1 row)

DROP TABLE t;
